spec/ndx-EXTRACT.extensions.yaml

```diff
@@ -3,232 +3,406 @@ groups:
   neurodata_type_inc: ImageSegmentation
   doc: EXTRACT configuration parameters
   attributes:
+  - name: version
+    dtype: text
+    doc: The version of EXTRACT.
+    required: false
   - name: preprocess
-    dtype: int32
-    doc: int32ean flag indicating data preprocessing before main EXTRACT function
+    dtype: bool
+    doc: Indicates whether or not data preprocessing was applied before main EXTRACT
+      function.
+    required: false
+  - name: pre_mask_on
+    dtype: bool
+    doc: Indicates whether or not an image mask was applied for preprocessing.
+    required: false
+  - name: pre_mask_radius
+    dtype: float64
+    doc: The radius of the image mask.
+    required: false
   - name: fix_zero_FOV_strips
-    dtype: int32
-    doc: int32ean flag. Find and fix spatial slices that are occasionally zero due
-      to frame registration during preprocessing.
+    dtype: bool
+    doc: Indicates whether or not find and fix spatial slices that are occasionally
+      zero due to frame registration during preprocessing.
+    required: false
   - name: medfilt_outlier_pixels
-    dtype: int32
-    doc: int32ean flag determining whether outlier pixels in the movie should be replaced
-      with their neighborhood median.
+    dtype: bool
+    doc: Determines whether outlier pixels in the movie should be replaced with their
+      neighborhood median.
+    required: false
   - name: skip_dff
-    dtype: int32
-    doc: int32ean flag. Skip Df/F calculation in preprocessing.
-  - name: baseline_quantile
-    dtype: float32
-    doc: Baseline quantile for Df/F calculation in preprocessing.
+    dtype: bool
+    doc: Indicates whether to skip Df/F calculation in preprocessing.
+    required: false
+  - name: second_df
+    dtype: float64
+    doc: The setting for second df calculation.
+    required: false
   - name: skip_highpass
-    dtype: int32
-    doc: int32ean flag. Skip highpass filtering in preprocessing.
+    dtype: bool
+    doc: Indicates whether to skip highpass filtering in preprocessing.
+    required: false
   - name: spatial_highpass_cutoff
-    dtype: float32
-    doc: 'Cutoff determining the strength of butterworth spatial filtering of the
-      movie. Values defined relative to the average cell radius. '
+    dtype: float64
+    doc: Cutoff determining the strength of butterworth spatial filtering of the movie.
+      Values defined relative to the average cell radius.
+    required: false
   - name: temporal_denoising
-    dtype: int32
-    doc: int32ean flag that determines whether to apply temporal wavelet denoising.
+    dtype: bool
+    doc: Determines whether to apply temporal wavelet denoising.
+    required: false
   - name: remove_background
-    dtype: int32
-    doc: int32ean flag that determines whether to subtract the (spatial) background
-      (largest spatiotemporal mode of the movie matrix).
+    dtype: bool
```

Comment:
Similar to above, this is encoded in MATLAB as double, but usage seems to be binary so fine to keep as bool here.